lib/parachain/validation_protocol.go

```diff
@@ -8,23 +8,199 @@ import (
 
 	"github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/lib/common"
+	parachaintypes "github.com/ChainSafe/gossamer/lib/parachain/types"
 	"github.com/ChainSafe/gossamer/pkg/scale"
 	"github.com/libp2p/go-libp2p/core/peer"
 )
 
 const MaxValidationMessageSize uint64 = 100 * 1024
 
-type ValidationProtocolV1 struct {
-	//	TODO: Implement this struct https://github.com/ChainSafe/gossamer/issues/3318
+type UncheckedSignedAvailabilityBitfield struct {
+	// The payload is part of the signed data. The rest is the signing context,
+	// which is known both at signing and at validation.
+	Payload scale.BitVec `scale:"1"`
```

Comment:
just making sure, no 0 index here?

Reply:
This is okay. It starts with 1. Search `scale:"0"` in the codebase, you won't find any entries.